### PR TITLE
Add force delete operation on server

### PR DESCRIPTION
**Pull request by Arun-LinkedIn**
Changes to place a delete tombstone on server even if blob is not present

### CODE REVIEW

File: ambry-server/src/main/java/com/github/ambry/server/AmbryRequests.java [hunk 1]
```diff
@@ -435,64 +434,86 @@ public void handleDeleteRequest(NetworkRequest request) throws IOException, Inte
...
-      } else if (e.getErrorCode() == StoreErrorCodes.TTL_Expired) {
-        metrics.ttlExpiredError.inc();
-      } else if (e.getErrorCode() == StoreErrorCodes.ID_Deleted) {
-        metrics.idDeletedError.inc();
-      } else if (e.getErrorCode() == StoreErrorCodes.Authorization_Failure) {
-        metrics.deleteAuthorizationFailure.inc();
+      if (e.getErrorCode() == StoreErrorCodes.ID_Not_Found && deleteRequest.shouldForceDelete()) {
+        try {
+          // If frontend forces a delete operation, place a tombstone even though blob is not present
+          Objects.requireNonNull(storeToDelete).forceDelete(Collections.singletonList(info));
```
Comment (justinlin-linkedin) [inline ambry-server/src/main/java/com/github/ambry/server/AmbryRequests.java:461]: if we are here, the `storeToDelete` will not be null.

Comment (justinlin-linkedin) [inline ambry-server/src/main/java/com/github/ambry/server/AmbryRequests.java:461]: one thing about the `forceDelete` is the the `MessageInfo` passed to this method can't have -1 lifeVersion. It has to be something meaningful. We have to change the MessageInfo to have 0 as lifeVersion.

Can you also add a comment to forceDelete method?

File: ambry-server/src/main/java/com/github/ambry/server/AmbryRequests.java [hunk 2]
```diff
@@ -435,64 +434,86 @@ public void handleDeleteRequest(NetworkRequest request) throws IOException, Inte
...
-      boolean logInErrorLevel = false;
-      if (e.getErrorCode() == StoreErrorCodes.ID_Not_Found) {
-        metrics.idNotFoundError.inc();
-      } else if (e.getErrorCode() == StoreErrorCodes.TTL_Expired) {
-        metrics.ttlExpiredError.inc();
-      } else if (e.getErrorCode() == StoreErrorCodes.ID_Deleted) {
-        metrics.idDeletedError.inc();
-      } else if (e.getErrorCode() == StoreErrorCodes.Authorization_Failure) {
-        metrics.deleteAuthorizationFailure.inc();
+      if (e.getErrorCode() == StoreErrorCodes.ID_Not_Found && deleteRequest.shouldForceDelete()) {
```
Comment (justinlin-linkedin) [inline ambry-server/src/main/java/com/github/ambry/server/AmbryRequests.java:458]: can we create a new method for this logic, something like `maybeForceDelete`.

**Review by justinlin-linkedin — COMMENTED** [review]
most parts LGTM. could you also add some comments.

File: ambry-server/src/main/java/com/github/ambry/server/AmbryRequests.java (hunk 2 above)
Comment (Arun-LinkedIn) [reply ambry-server/src/main/java/com/github/ambry/server/AmbryRequests.java:458]: Sure. Created a new method for this logic.

File: ambry-server/src/main/java/com/github/ambry/server/AmbryRequests.java (hunk 1 above)
Comment (Arun-LinkedIn) [reply ambry-server/src/main/java/com/github/ambry/server/AmbryRequests.java]: Yeah.. true.

Comment (Arun-LinkedIn) [reply ambry-server/src/main/java/com/github/ambry/server/AmbryRequests.java]: Sure. Added comments to the forceDelete method that the lifeVersion of input should be a valid value.

**Review by justinlin-linkedin — COMMENTED** [review]
LGTM, please add some test cases for the new code.

**Review by justinlin-linkedin — APPROVED** [review]

**Pull request merged**